src/headerChainSync/HeaderChainProvider.js

```diff
@@ -4,40 +4,28 @@
  * @module HeaderChainProvider
  */
 const { SpvChain } = require('@dashevo/dash-spv');
-const Api = require('../');
-const MNDiscovery = require('../MNDiscovery/index');
-const config = require('../config');
-const constants = require('../constants');
+const range = require('lodash/range');
 
 /**
  * This class syncs the header chain in a parallel manner across masternodes
  */
 class HeaderChainProvider {
   /**
    * @param {string} network
-   * @param {SimplifiedMNListEntry[]} seeds
-   * @param {number} DAPIPort
+   * @param {DAPIClient} api
+   * @param {int} mnListLength
    */
-  constructor(network, seeds, DAPIPort = config.Api.port) {
+  constructor(network, api, mnListLength) {
```

Comment:
I would move `network` and `mnListLength` to the options and give them default values.